src/plugins/ui/renderer.ts

```diff
@@ -607,6 +607,10 @@ export class RendererPlugin extends UIPlugin {
       } else {
         nextColIndex = this.findNextEmptyCol(col, right, row);
         previousColIndex = this.findPreviousEmptyCol(col, left, row);
+
+        // Prevent overflow over borders
+        previousColIndex = box.border?.left ? col : previousColIndex;
```

Comment:
A bit of  'économie de bout de ficelle' but you could
```suggestion
        previousColIndex = box.border?.left ? col : this.findPreviousEmptyCol(col, left, row);
        nextColIndex = box.border?.right ? col : this.findNextEmptyCol(col, right, row);
```
no need to look for the empty cols if there's a border ;-)